### DISCUSSION TITLE
Access context values inside of getLayout

### DISCUSSION THREAD

**Discussion by jamesvclements** [vercel/next.js#41394] [Help] [answered]
Is there a way to use a hook or access the component's hook values inside of the `getLayout` function?

![index js — microjournal-Thursday-October-13-2022-12 51 21PM@2x](https://user-images.githubusercontent.com/20052710/195658693-18febcb8-22e0-45c6-830e-8fe09a063d55.png)

Or, is there a better way I should be approaching this usecase?

I'm trying to have a dashboard that shows the login component `<Auth />` at the root page when the user isn't logged in, otherwise it shows their `<Profile />`. This currently works with the `<Home />` page shown above, but now I'm trying to add a `<SideNav />` into the authenticated routes.

I thought I could do this with `getLayout` but I need to do it conditionally for the `<Home` page whether `session` exists or not.

**Comment by jamesvclements** [comment, marked as answer]
Would it make more sense to just do this inside my main `<Layout>` component?

![layout js — microjournal-Thursday-October-13-2022-01 21 05PM@2x](https://user-images.githubusercontent.com/20052710/195664025-4a80b4a6-168e-4d4d-beb6-d75eafd3b6fd.png)

**Reply by icyJoseph** [reply]
Yes, because the `getLayout` is not a React component really, so defining react logic inside of it is unstable. Look at the function signature there's no `children` prop, but a page argument whole. 

Read here, https://kentcdodds.com/blog/dont-call-a-react-function-component, near the end of the posts.

**Reply by samithaf** [reply]
It's in React docs as well. 

Don’t call Hooks from regular JavaScript functions. Instead, you can:

✅ Call Hooks from React function components.
✅ Call Hooks from custom Hooks (we’ll learn about them [on the next page](https://reactjs.org/docs/hooks-custom.html)).

https://reactjs.org/docs/hooks-rules.html

**Reply by icyJoseph** [reply]
It's funny because in practice you won't see getLayout on the React tree, for example through DevTools, and the React stuff defined within it falls to CustomApp in this case, but yeah, don't do it :D

**Reply by jamesvclements** [reply]
@icyJoseph @samithaf makes perfect sense. Thank you both!
